### PR TITLE
Paint more features

### PR DESCRIPTION
**Pull request by LanesGood**
~PR to add more features to draw in app. First test feature is `tag = amenity`, though I'm unsure whether this is called correctly in the `filters`. The amenities rendered are also not yet clickable.~

~@developmentseed/observe I'd love to know more about how to set up the filters correctly, and if there are other specific features we should be adding. Amenities seems to capture quite a lot of the missing tags, though the only way I'm verifying this is by browsing an area in Observe and OpenStreetMap at the same time.~

In screenshot below, the light primary purple color represents `amenity = parking lot`, while the secondary red color is an edited polygon.

image

### CODE REVIEW

**Comment by geohacker** [comment]
@LanesGood this looks good to me, I think we have two ways to handle this:
* go through the feature type styles in iD here https://github.com/openstreetmap/iD/tree/develop/css and add them as layers. We can simplify many along the way.
* add a generic line and polygon layer with a standard style that renders everything. 

To make the layers clickable, you can just add the name here https://github.com/developmentseed/observe/blob/paint-more-features/app/screens/Explore.js#L159

**Comment by LanesGood** [comment]
@geohacker thanks for the direction on iD styles, that helps me figure out what should be included! I added the generic line and polygon layers as you suggested.

@kamicut helped me as well with the filter expression language. He also helped bring up the relevant question "Why style everything twice?"

I'm still working through some of the base feature styles in iD, but I don't think many of them are needed. With the generic clickable polygon. multipolygon and line, users should be able to interact with and see the bounds of any feature... The question is, should they? This may make for a distracting and difficult experience where any and every screen tap results in the presence of feature outlines, preview cards with feature info, etc. I think this needs more in depth testing, and styles should be added according to feature requirements.

**Comment by LanesGood** [comment]
PR ready for review, although there are still some issues to resolve:
- Some double styles being applied; buildings that are also amenities, etc. 
- Some features (buildings, paths) captured within large amenity areas like Universities are difficulty to see. I'm not quite sure how to avoid applying double styles here either
- Not all features captured within areas are clickable
- Some larger boundaries on natural areas still not being captured despite the `boundary` and `natural` tags, resulting in continued intersections with unseen ways/areas when in way/editing mode. This may just be a result of having many parks/natural areas that have both lines and multi-polygon definitions for the boundaries, but I thought this would be captured

**Comment by geohacker** [comment]
This looks good to me. Though, I see the problem of overcrowded map and almost impossible to hide the drawer. 

> "Why style everything twice?"

We definitely don't want to style things twice. That's not the intention at all with the generic layers. The reason why I suggested that is because if we don't have the time to go through individual presets then the most easiest would be to add the generic style layers. Another approach is to make them only visible in the Edit Way or Add Way modes.

@LanesGood 
* We don't want to style coastlines and boundaries - let's remove
* Some features that are make sense to show 
    * [water features](https://github.com/openstreetmap/iD/blob/develop/css/45_waterways.css)
    * [railways](https://github.com/openstreetmap/iD/blob/develop/css/40_railways.css)
    * [area features](https://github.com/openstreetmap/iD/blob/develop/css/25_areas.css#L25)

Let's not style everything right now. What if we load the generic line and polygon styles in the Edit and Add Way mode?

**Comment by geohacker** [comment]
@LanesGood in the last commit, I tried to remove double-styling by making the filters a bit more specific. I'd say this is the best approach. I also commented out boundaries and coastlines for now.

Also, the compass settings where throwing a warning for me on iOS and it wasn't working, so I hid them for now. Does this work for you?

**Comment by LanesGood** [comment]
@geohacker thanks for setting up the exclusions of already rendered styles. 

I'm happy to continue on the railways, water and area features in a new branch once this is merged in.

I like the idea of only rendering the generic fill and polygon layers in add/edit way mode, and attempted to set this up. Any idea on why the below won't work?
image

File: app/screens/Explore.js [hunk 1]
```diff
@@ -686,15 +725,26 @@ class Explore extends React.Component {
...
                       <MapboxGL.FillLayer id='leisure' filter={filters.leisure} style={style.osm.leisure} minZoomLevel={16} />
+                      <MapboxGL.FillLayer id='natural' filter={filters.natural} style={style.osm.natural} minZoomLevel={16} />
                       <MapboxGL.LineLayer id='selectedFeatures' filter={filters.selectedFeatures} style={style.osm.selectedFeatures.lines} minZoomLevel={16} />
                       <MapboxGL.CircleLayer id='iconHalo' style={style.osm.iconHalo} minZoomLevel={16} filter={filters.iconHalo} />
                       <MapboxGL.CircleLayer id='iconHaloSelected' style={style.osm.iconHaloSelected} minZoomLevel={16} filter={filters.iconHaloSelected} />
                       <MapboxGL.SymbolLayer id='pois' style={style.osm.icons} filter={filters.pois} />
                     </MapboxGL.ShapeSource>
+                    {
+                      (mode === modes.EDIT_WAY || mode === modes.ADD_WAY) &&
+                        <MapboxGL.ShapeSource id='geojsonSource' shape={geojson}>
```
Comment (geohacker) [inline app/screens/Explore.js:743]: @LanesGood adding the geojson source again on the map in the EDIT / ADD WAY mode will be a bit inefficient. I'd recommend setting a `visibility` on the layers and switching them in the mode.

**Comment by geohacker** [comment]
@LanesGood i pushed a fix. Let me know what you think and feel free to merge this if it looks ok!

**Comment by LanesGood** [comment]
@geohacker looks great - thanks!

**Pull request merged**